t/03_collapse.t

```diff
@@ -7,27 +7,40 @@ use Data::NestedParams;
 
 cmp_deeply(
     collapse_nested_params({ a => undef }),
-    { a => undef },
+    [ a => undef ],
 );
 
 cmp_deeply(
     collapse_nested_params({ a => 3 }),
-    { a => 3 },
+    [ a => 3 ],
 );
 
 cmp_deeply(
     collapse_nested_params({ a => [qw(x y z)] }),
-    { 'a[]' => 'x', 'a[]' => 'y', 'a[]' => 'z'},
+    [ 'a[]' => 'x', 'a[]' => 'y', 'a[]' => 'z' ],
 );
 
 cmp_deeply(
     collapse_nested_params({ a => { b => 3 } }),
-    { 'a[b]' => '3'},
+    [ 'a[b]' => '3' ],
 );
 
 cmp_deeply(
-    collapse_nested_params({'x' => {'y' => [{'z' => '1','w' => 'a'},{'z' => '2','w' => '3'}]}}),
-    {'x[y][][z]',1,'x[y][][w]','a','x[y][][z]',2,'x[y][][w]',3}
+    collapse_nested_params({
+        'x' => {
+            'y' => [
+                {'z' => '1','w' => 'a'},
+                {'z' => '2','w' => '3'}
+            ]
+        }
+    }),
+    # NOTE: Order here depends on perl hashing function
+    [
```

Comment:
The ordering of elements in perl hashes is not deterministic, even when given the same input. So I think it would be better to test this using something like `Test::Dump::cmp_bag`. Or just check for each element individually.